### PR TITLE
Update instructions to install extensions

### PR DESCRIPTION
**Pull request by fengzhou-msft**
Update instructions for using the dynamic extension install feature.

### CODE REVIEW

File: docs-ref-conceptual/azure-cli-extensions-overview.md [hunk 1]
```diff
@@ -43,6 +45,22 @@ az extension add --source <URL-or-path>
 Once an extension is installed, it's found under the value of the `$AZURE_EXTENSION_DIR` shell variable. If this variable is unset, by default the value is `$HOME/.azure/cliextensions` on
 Linux and macOS, and `%USERPROFILE%\.azure\cliextensions` on Windows.
 
+### Install extensions automatically
+
+Azure CLI also has the ability to install an extension dynamically based on the command you run if the command is from an extension that is not installed yet. You can enable the feature through configuration:
+```azurecli-interactive
+az config set extension.use_dynamic_install=yes_prompt
```
Comment (yungezz) [inline docs-ref-conceptual/azure-cli-extensions-overview.md:52]: how about list the 3 options with description on behavior of each option

File: docs-ref-conceptual/azure-cli-extensions-overview.md [hunk 2]
```diff
@@ -43,6 +45,22 @@ az extension add --source <URL-or-path>
 Once an extension is installed, it's found under the value of the `$AZURE_EXTENSION_DIR` shell variable. If this variable is unset, by default the value is `$HOME/.azure/cliextensions` on
 Linux and macOS, and `%USERPROFILE%\.azure\cliextensions` on Windows.
 
+### Install extensions automatically
+
+Azure CLI also has the ability to install an extension dynamically based on the command you run if the command is from an extension that is not installed yet. You can enable the feature through configuration:
```
Comment (dbradish-microsoft) [inline docs-ref-conceptual/azure-cli-extensions-overview.md:50]: When you run an extension command that is not installed, the Azure CLI can recognize the command you run, and automatically install the extension for you.  This feature, referred to as **dynamic install**, is enabled through configuration.

File: docs-ref-conceptual/azure-cli-extensions-overview.md [hunk 3]
```diff
@@ -43,6 +45,22 @@ az extension add --source <URL-or-path>
...
+```azurecli-interactive
+az config set extension.use_dynamic_install=yes_prompt
+```
+To use for automation without prompt:
```
Comment (dbradish-microsoft) [inline docs-ref-conceptual/azure-cli-extensions-overview.md:54]: Use the following syntax to enable dynamic install without a prompt.

File: docs-ref-conceptual/azure-cli-extensions-overview.md [hunk 4]
```diff
@@ -43,6 +45,22 @@ az extension add --source <URL-or-path>
...
+```azurecli-interactive
+az config set extension.use_dynamic_install=yes_without_prompt
+```
+
+By default, the command stops after the needed extension is installed and you need to rerun the command. You can make the command continue to run after the extension is installed:
```
Comment (dbradish-microsoft) [inline docs-ref-conceptual/azure-cli-extensions-overview.md:59]: By default, an extension command that prompts dynamic install will not continue to run.  You can change the default behavior and make the command continue by setting the **run_after_dynamic_install** property to **yes**.

File: docs-ref-conceptual/azure-cli-extensions-overview.md (hunk 1 above)
Comment (fengzhou-msft) [reply docs-ref-conceptual/azure-cli-extensions-overview.md:52]: Listed all 3 options.

File: docs-ref-conceptual/azure-cli-extensions-overview.md [hunk 5]
```diff
@@ -43,6 +45,28 @@ az extension add --source <URL-or-path>
 Once an extension is installed, it's found under the value of the `$AZURE_EXTENSION_DIR` shell variable. If this variable is unset, by default the value is `$HOME/.azure/cliextensions` on
 Linux and macOS, and `%USERPROFILE%\.azure\cliextensions` on Windows.
 
+### Install extensions automatically
+
+When you run an extension command that is not installed, the Azure CLI can recognize the command you run, and automatically install the extension for you. This feature, referred to as **dynamic install**, is enabled through configuration.
```
Comment (fengzhou-msft) [inline docs-ref-conceptual/azure-cli-extensions-overview.md:50]:
```suggestion
When you run an extension command that is not installed, the Azure CLI can recognize the command you run, and automatically install the extension for you starting from version `2.10.0`. This feature, referred to as **dynamic install**, is enabled through configuration.
```

**Review by yungezz — APPROVED** [review]
LGTM

**Pull request merged**